vms/platformvm/service.go

```diff
@@ -1191,6 +1191,21 @@ func (s *Service) AddValidator(req *http.Request, args *AddValidatorArgs, reply
 		zap.String("method", "addValidator"),
 	)
 
+	tx, changeAddr, err := s.buildAddValidatorTx(args)
+	if err != nil {
+		return fmt.Errorf("couldn't create tx: %w", err)
+	}
+
+	reply.TxID = tx.ID()
+	reply.ChangeAddr, err = s.addrManager.FormatLocalAddress(changeAddr)
+	if err != nil {
+		return fmt.Errorf("couldn't format address: %w", err)
+	}
+
+	return s.vm.issueTx(req.Context(), tx)
```

Comment:
I was slightly concerned providing `req.Context()` here in the X-chain because these contexts can be cancelled by the client invoking this API... We also document that if `AppSender` returns an error it should be treated as FATAL... However for the rpcchainvm the gRPC server may return an error if the provided context is cancelled... We never expect the platformvm to run in to this situation... but wanted to note this so that we could talk about it

Reply:
> We also document that if AppSender returns an error it should be treated as FATAL

I remember this being an invariant but I never understood why. Why can't a sender just log the error and move on? If we're letting the request context be cancelled by the client which is a legitimate use-case I feel like we shouldn't be expected to fatal.

Reply:
The comment was a prior invariant before we passed the context over the gRPC. Feel free to make a PR to remove that comment